### ISSUE TITLE
Creation of custom quarantine path fails

### ISSUE THREAD

**Issue by osworx** [Studio-42/elFinder#3059]
Dev environement WIN10, XAMPP, latest elFinder version from here.

Custom quarantine path (e,g,): d:/server/image/quarantine
leads to (debug output):

**Driver "elFinderVolumeTrash" : Root folder does not exist.**
and
**WARNING: mkdir(): No such file or directory in elFinderVolumeLocalFileSystem.class.php line 197.**

Looking at this line:
**$this->quarantine = $this->_abspath($this->options['quarantine']);**
and checking the options value and generated value of the function **_abspath** (line before)
**$this->quarantine = $this->_abspath($this->options['quarantine']);**

gives me that path:

> D:\server\image\D:\server\image\quarantine

which comes from the function **_absPath** and the line there:
**return $this->_joinPath($this->root, $path);**

**_joinPath** return this: 
**return rtrim($dir, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR . $name;**

which leads to the wrong path above which of course cannot be created.

**Comment by nao-pon** [comment]
@osworx Please show L.190-191
```php
            if (is_dir($this->options['quarantine'])) {
                if (is_writable($this->options['quarantine'])) {
```
If the given path is a directory and is writable, use that path.

**Comment by osworx** [comment]
Checking these lines and closing this issue does not solve the bug.
Line 190:
> if (is_dir($this->options['quarantine'])) {

So good, so fine - not valid if a custom dir is defined (see my question).
Lead to line 196:

> $this->quarantine = $this->_abspath($this->options['quarantine']);

which gives us the wrong path (as already mentioned):

> D:\server\image\D:\server\image\quarantine

Following line 197 is also no help in this case:

> if ((!is_dir($this->quarantine) && !mkdir($this->quarantine)) || !is_writable($this->quarantine)) {

because 
> D:\server\image\D:\server\image\quarantine

is useless.

So, defining a custom path (and not exisiting therefore at the first access), should create it.
Using `mkdir('my_custom_path', 0755, true)` and checking the function **_abspath** would be helpfully ..

**Comment by nao-pon** [comment]
@osworx I understand. This is because the _abspath () function cannot handle Windows server specific drive letters. I will fix this.

**Comment by nao-pon** [comment]
I fixed in 969b61c96c89f7d873e482718b03865e06c95cc1. Please check it.
